src/core/mac/mac.cpp

```diff
@@ -261,14 +289,15 @@ void Mac::StartEnergyScan(void)
     }
     else
     {
-        otError error = otPlatRadioEnergyScan(&GetInstance(), mScanChannel, mScanDuration);
+        SuccessOrExit(error = otPlatRadioEnergyScan(&GetInstance(), mScanChannel, mScanDuration));
+    }
 
-        if (error != OT_ERROR_NONE)
-        {
-            // Cancel scan
-            mEnergyScanHandler(mScanContext, NULL);
-            FinishOperation();
-        }
+exit:
+
+    if (error != OT_ERROR_NONE)
+    {
+        mEnergyScanHandler(mScanContext, NULL);
```

Comment:
See comment above.

Reply:
Changed the logic to use the task in the new pushed commit.